identity/src/main/java/com/stripe/android/identity/analytics/IdentityAnalyticsRequestFactory.kt

```diff
@@ -25,34 +28,77 @@ internal class IdentityAnalyticsRequestFactory @Inject constructor(
         origin = ORIGIN
     )
 
-    private fun additionalParamWithEventMetadata(vararg pairs: Pair<String, *>) =
-        mapOf(
+    private fun additionalParamWithEventMetadata(vararg pairs: Pair<String, *>): Map<String, Any> {
+        val metadataMap = mutableMapOf(*pairs)
+        verificationPage?.livemode?.let { liveMode ->
+            metadataMap[PARAM_LIVE_MODE] = liveMode
+        }
+
+        return mapOf(
             PARAM_VERIFICATION_SESSION to args.verificationSessionId,
-            PARAM_EVENT_META_DATA to
-                mutableMapOf(
-                    *pairs
-                ).also {
-                    verificationPage?.let {
-                        PARAM_LIVE_MODE to it.livemode
-                    }
+            PARAM_EVENT_META_DATA to metadataMap
+        )
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private suspend fun maybeLogExperimentAndSendLog(
+        eventName: String,
+        additionalParams: Map<String, Any> = mapOf()
+    ) {
+        verificationPage?.let { verificationPage ->
+            val experiments = verificationPage.experiments
+            val userSessionId = verificationPage.userSessionId
+            val metaDatas = additionalParams.getOrDefault(PARAM_EVENT_META_DATA, null) as Map<String, Any>?
+
+            for (exp: VerificationPageStaticContentExperiment in experiments) {
+                if (exp.matches(eventName, metaDatas)) {
```

Comment:
```suggestion
            experiments.any { it.matches(eventName, metaDatas)) {
```

Reply:
repalced with filter as we need to actually get the experient matched